packages/grafana-ui/src/components/uPlot/config/UPlotScaleBuilder.ts

```diff
@@ -11,12 +10,21 @@ export interface ScaleProps {
 
 export class UPlotScaleBuilder extends PlotConfigBuilder<ScaleProps, Scale> {
   getConfig() {
-    const { isTime, scaleKey, min, max } = this.props;
-    const range = isNumber(min) && isNumber(max) ? [min, max] : undefined;
+    const { isTime, scaleKey } = this.props;
+    if (isTime) {
+      return {
+        [scaleKey]: {
+          time: true, // no explicit ranges for time?
+        },
+      };
+    }
     return {
       [scaleKey]: {
-        time: !!isTime,
-        range,
+        range: (u: uPlot, initMin: number, initMax: number) => {
+          const { min, max } = this.props;
+          const [smin, smax] = uPlot.rangeNum(initMin, initMax, 0.05 as any, true);
```

Comment:
if your know that you have an explicitly-set min or max, it probably makes sense to pass these to the rangeNum instead of the dataMin/Max, this way it expands the padding relative to the final range rather than the data range.

```
uPlot.rangeNum(min ?? initMin, max ?? initMax, 0.05 as any, true);
```

also 0.05 padding factor is pretty tight, the default is 0.1, but your call.